libs/MotionKit/source/MotionKit.cpp

```diff
@@ -0,0 +1,115 @@
+// Leka - LekaOS
+// Copyright 2022 APF France handicap
+// SPDX-License-Identifier: Apache-2.0
+
+#include "MotionKit.h"
+
+#include "MathUtils.h"
+#include "ThisThread.h"
+
+using namespace leka;
+using namespace std::chrono_literals;
+
+void MotionKit::init()
+{
+	_event_loop.registerCallback([this] { run(); });
+}
+
+void MotionKit::stop()
+{
+	_stabilisation_requested  = false;
+	_target_not_reached		  = false;
+	_rotate_x_turns_requested = false;
+	_motor_left.stop();
+	_motor_right.stop();
+	_event_loop.stop();
```

Comment:
I would stop first, as this is what's important, then assign the values

```suggestion
	_motor_left.stop();
	_motor_right.stop();
	_event_loop.stop();
	
	_stabilisation_requested  = false;
	_target_not_reached		  = false;
	_rotate_x_turns_requested = false;
```